crates/bevy_scene/src/dynamic_scene.rs

```diff
@@ -52,7 +52,11 @@ impl DynamicScene {
                 let reflect_component = world
                     .components()
                     .get_info(component_id)
-                    .and_then(|info| type_registry.get(info.type_id().unwrap()))
+                    .and_then(|info| {
+                        type_registry.get(info.type_id().unwrap_or_else(|| {
+                            panic!("Component {:?} has no `type_id`", info.name())
```

Comment:
This is probably misleading. As I understand it, this is likely to be the result of a failure to register the component type in the registry (but please double-check!).

Reply:
This was definitely the message I had the least confidence in. Will investigate, thanks!

Reply:
I think the error you are referring to would be an error in `type_registry.get()`, not in `info.type_id()`. I don't know why `ComponentDescriptor.type_id` is of type Option<TypeId>. As far as I understand we should be able to change it to just `TypeId`.
https://github.com/bevyengine/bevy/blob/c4f132afbfe5688afd13f9b05040dfdf98b65489/crates/bevy_ecs/src/component.rs#L166-L177